csrc/multidevice/utils.h

```diff
@@ -123,9 +123,9 @@ int64_t requestedNumberOfDevices(Fusion*);
 void unshard(Fusion*);
 void unshard(TensorView*);
 
-// Returns the index of the a sharded axis if none return -1.
-// TODO: Assumes no merges/splits on sharded axis.
-int64_t getShardedAxis(TensorView*);
+// Returns the index of the sharded logical axis corresponding to
+// `parallel_type`. If `tv` isn't sharded on the parallel type, returns -1.
```

Comment:
IIUC, when sharding a split and/or transposed loop's domain axis, then this function returns the (unique) logical domain's axis that produces that axis.

I find the comment misleading here, and would suggest to make it more explicit, e.g,
```suggestion
// Returns the index of the logical axis which is the producer of the loop axis sharded with
// `parallel_type`. If `tv` isn't sharded on the parallel type, returns -1.
```

In general, I am skeptical about how this function can be conveniently used without returning also the information on how to transform this logical axis to the logical sharded axis...